contracts/oracle/src/contract/oracle/mod.rs

```diff
@@ -129,6 +129,15 @@ where
             .take(max_count.try_into()?)
             .collect::<ContractResult<Vec<Addr>>>()?;
 
+        #[cfg(debug_assertions)]
+        {
+            use std::collections::HashSet;
+
+            let set: HashSet<&Addr> = HashSet::from_iter(subscribers.as_slice());
+
+            debug_assert_eq!(set.len(), subscribers.len());
+        }
+
```

Comment:
please extract this chunk into a function ... we shall keep the function body succint